### ISSUE TITLE
Package manager

### ISSUE THREAD

**Issue by brendanzab** [pikelet-lang/pikelet#178]
It would be neat to have a way to define Pikelet packages!

I'm also thinking it would also be neat to be able to define packages using Pikelet records!

```
record {
    name = "prelude";
    version = "0.1.0";
    copyright = "ASL-2.0";
    dependencies = [];
}
```

Having custom literals or something, perhaps like in [relit](https://github.com/cyrus-/relit) might make this nicer 🤔

## TODO List

- [ ] add a `pikelet-package` crate
- [ ] ???

**Comment by brendanzab** [comment]
Might be interesting to have a look at [cargo](https://github.com/rust-lang/cargo/) to see what ideas we can steal. I'd also like to pick Yehuda Katz's brain on what he would do differently. I think he was doing work here: [argon-package](https://github.com/wycats/argon/tree/master/crates/argon-package). It would also be interesting to look at Nix too.

**Comment by boomshroom** [comment]
That right there is valid Nix syntax, though `record` would be parsed as a function rather than a keyword because it doesn't have to disambiguate between types and values (dynamically typed). It wouldn't know what to do with it, but it would parse and it wouldn't be too hard to translate that into a valid Niix derivation.

**Comment by brendanzab** [comment]
Yeah, I think being able to abstract Pikelet to Nix, Dhall, JSON, etc. would be kind of neat! It would be cool if one could statically verify that a bit of Pikelet code could be transformed into one of those formats. Not sure how hard that would be though.

**Comment by brendanzab** [comment]
I'm definitely interested in languages that can be their own data formats - like REBOL and Lisp.
